jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricherTest.java

```diff
@@ -173,4 +172,41 @@ public void testCreateWithDefaultsAndExistingRouteWithMatchingNameInBuilderInOpe
             .extracting("metadata.name", "spec.host", "spec.to.kind", "spec.to.name", "spec.port.targetPort.intVal")
             .contains("test-svc", "example.com", null, null, 1337);
     }
+
+    @Test
+    public void testEnrichNoTls(){
+        // Given
+        new RouteEnricher(context).create(PlatformMode.openshift, klb);
+        // When
+        new RouteEnricher(context).enrich(PlatformMode.openshift, klb);
+        // Then
+        assertThat(klb.build().getItems())
+                .hasSize(2)
+                .extracting("kind")
+                .containsExactly("Service", "Route");
+
+        assertThat(klb.build().getItems().stream().filter(h -> h.getKind().equals("Route")).findFirst().orElse(null))
+                .extracting("spec.tls")
+                .containsNull();
+
+    }
+
+    @Test
+    public void testEnrichWithTls(){
+        // Given
+        properties.put("jkube.openshift.generateRoute.tls.termination", "edge");
+        properties.put("jkube.openshift.generateRoute.tls.insecure_edge_termination_policy", "Allow");
+        new RouteEnricher(context).create(PlatformMode.openshift, klb);
+        // When
+        new RouteEnricher(context).enrich(PlatformMode.openshift, klb);
```

Comment:
Umm, I think it's just my personal opinion. But would it be nice to do it using only single instance of RouteEnricher?

Reply:
@rohanKanojia  I agree. I have removed the enrich method in RouteEnricher entirely, so I can keep the structure of tests and method invocations as-is.